arangod/Agency/ResignLeadership.cpp

```diff
@@ -0,0 +1,480 @@
+      addPreconditionUnchanged(*pending, failedServersPrefix, failedServers);
+      addPreconditionUnchanged(*pending, cleanedPrefix, cleanedServers);
+    }
+  }  // array for transaction done
+
+  // Transact to agency
+  write_ret_t res = singleWriteTransaction(_agent, *pending, false);
+
+  if (res.accepted && res.indices.size() == 1 && res.indices[0]) {
+    LOG_TOPIC("deadd", DEBUG, Logger::SUPERVISION) << "Pending: Clean out server " + _server;
+
+    return true;
+  }
+
+  LOG_TOPIC("deade", INFO, Logger::SUPERVISION)
+      << "Precondition failed for starting ResignLeadership job " + _jobId;
+
+  return false;
+}
+
+bool ResignLeadership::scheduleMoveShards(std::shared_ptr<Builder>& trx) {
+  std::vector<std::string> servers = availableServers(_snapshot);
+
+  Node::Children const& databases = _snapshot.hasAsChildren("/Plan/Collections").first;
+  size_t sub = 0;
+
+  for (auto const& database : databases) {
+    // Find shardsLike dependencies
+    for (auto const& collptr : database.second->children()) {
+      auto const& collection = *(collptr.second);
+
+      if (collection.has("distributeShardsLike")) {
+        continue;
+      }
+
+      for (auto const& shard : collection.hasAsChildren("shards").first) {
+        // Only shards, which are affected
+        int found = -1;
+        int count = 0;
+        for (auto const& dbserver : VPackArrayIterator(shard.second->slice())) {
+          if (dbserver.copyString() == _server) {
+            found = count;
+            break;
+          }
+          count++;
+        }
+        if (found == -1) {
+          continue;
+        }
+
+        bool isLeader = (found == 0);
+
+        if (isLeader) {
+
+          std::string toServer = Job::findNonblockedCommonHealthyInSyncFollower(
+            _snapshot, database.first, collptr.first, shard.first);
+
+          if (toServer.empty()) {
+            continue ; // can not resign from that shard
+          }
+
+          MoveShard(_snapshot, _agent, _jobId + "-" + std::to_string(sub++),
+                  _jobId, database.first, collptr.first, shard.first, _server,
+                  toServer, isLeader, false)
+            .create(trx);
+
+        } else {
+          // Intentionally do nothing. RemoveServer will remove the failed follower
+          LOG_TOPIC("deadf", DEBUG, Logger::SUPERVISION) <<
+            "Do nothing for resign leadership of follower of the collection " << collection.hasAsString("id").first;
+          continue ;
+        }
+      }
+    }
+  }
+
+  return true;
+}
+
+bool ResignLeadership::checkFeasibility() {
+  std::vector<std::string> availServers = availableServers(_snapshot);
+
+  // Minimum 1 DB server must remain:
+  if (availServers.size() == 1) {
+    LOG_TOPIC("deaa0", ERR, Logger::SUPERVISION)
+        << "DB server " << _server << " is the last standing db server.";
+    return false;
+  }
+
+  // Check that each shard has an in sync follower
+  // TO BE IMPLEMENTED
```

Comment:
No, this is intentional. The `resignLeadership` job should always be fast, it cannot help a customer who has chosen replicationFactor 1. We should remove the "TO BE IMPLEMENTED" and say so in a comment here.